File: 0977/solution.py
class Solution:
    def sortedSquares(self, nums: list[int]) -> list[int]:
        l, r = 0, len(nums) - 1
        res = []
        while l < r:
            if abs(nums[l]) < abs(nums[r]):
                res.append(nums[r] ** 2)
                r -= 1
            else:
                res.append(nums[l] ** 2)
                l += 1

        res.append(nums[l] ** 2)
        return res[::-1]
        # time O(n)
        # space O(1)

File: 0977/solution.java
class Solution {
   public int[] sortedSquares(int[] nums) {
      int l = 0;
      int r = nums.length - 1;
      int[] res = new int[nums.length];
      for (int i = 0; i < nums.length; i++) {
         int left = (int) Math.pow(nums[l], 2);
         int right = (int) Math.pow(nums[r], 2);
         if (left > right) {
            res[res.length - i - 1] = left;
            l++;
         } else {
            res[res.length - i - 1] = right;
            r--;
         }
      }
      return res;

   }
}